src/iso15118/io/socket_helper.cpp

```diff
@@ -31,10 +33,18 @@ bool get_first_sockaddr_in6_for_interface(const std::string& interface_name, soc
             continue;
         }
 
-        if (interface_name.compare(current_if->ifa_name) != 0) {
+        if (interface_name != "auto" && interface_name.compare(current_if->ifa_name) != 0) {
             continue;
         }
 
+        if (interface_name == "auto") {
+            inet_ntop(AF_INET6, &current_if->ifa_addr->sa_data, buffer, sizeof(buffer));
+            if (strstr(buffer, "fe80") == nullptr) {
```

Comment:
this might fail, if `fe80` is not at the beginning of the address (could be also somewhere else

Reply:
strstr returns a pointer where the pattern was found. Even in the middle or end of the string. Is that the problem for you? I don't quite understand where you see the problem

Reply:
Now the function is using `IN6_IS_ADDR_LINKLOCAL` instead of searching `fe80` in a string